### PR TITLE
Week 3 Business site Karolin Albinsson

### PR DESCRIPTION
**Pull request by karolinalbinsson**

### CODE REVIEW

File: code/style.css [hunk 1]
```diff
@@ -1,7 +1,482 @@
...
+.hero-video-wrap{
+  position: sticky;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+}
+
+.hero-video{
+   background-size: cover;
+    height: auto;
```
Comment (EmOlsz) [inline code/style.css:22]: Height is `auto` by default, no need to type it once again

File: code/style.css [hunk 2]
```diff
@@ -1,7 +1,482 @@
...
+    z-index: -1;
+}
+
+.hero-normal-titles-wrapper{
+  z-index: 1;
+  position: absolute;
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    padding: 3%;
```
Comment (EmOlsz) [inline code/style.css:36]: with paddings and margins it's far better to use `px` units

File: code/style.css [hunk 3]
```diff
@@ -1,7 +1,482 @@
...
+}
+
+.hero-title-normal{
+  font-size: 6em;
+  color: #c2b3a8;
+  font-family: 'Big Shoulders Display', cursive;
```
Comment (EmOlsz) [inline code/style.css:42]: Please remember about fallback font

File: code/style.css [hunk 4]
```diff
@@ -1,7 +1,482 @@
...
+
+.hero-small-title-normal{
+  font-size: 4em;
+  color: #c2b3a8;
+  font-family: Sacramento, cursive;
+  margin: 0;
+}
+
+.hero-mobile {
+  height: 40vh;
```
Comment (EmOlsz) [inline code/style.css:54]: Great job with this class :)

File: code/style.css [hunk 5]
```diff
@@ -1,7 +1,482 @@
...
+  background-image: url("./pics/jungleView.jpg");
+  background-position: center;
+  background-size: cover;
+  background-repeat: no-repeat;
+
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  font-family: 'Big Shoulders Display', cursive;
+  margin-bottom: 4vh;
```
Comment (EmOlsz) [inline code/style.css:64]: Once again, it's better to use `px`

File: code/style.css [hunk 6]
```diff
@@ -1,7 +1,482 @@
...
+  display: none;
+}
+
+.hero-header-mobile{
+  display: flex;
+  align-items: center;
+  flex-direction: column;
+}
+
+#header {
```
Comment (EmOlsz) [inline code/style.css:75]: We shouldn't style by `id`. Just add another class to that element

File: code/style.css [hunk 7]
```diff
@@ -1,7 +1,482 @@
...
+.hero-title-mobile{
+  background: #e8f3f8b8;
+  -webkit-background-clip: text;
+  -webkit-text-fill-color: transparent;
+  font-size: 2.8em;
+  margin-top: 0;
+}
+
+.hero-small-title-mobile{
+  background: #e8f3f8b8;
```
Comment (EmOlsz) [inline code/style.css:90]: Hmm, there is no color like that, there are two characters too much

File: code/style.css [hunk 8]
```diff
@@ -1,7 +1,482 @@
...
+.hero-mobile, .signup-form-wrap, .accordion-Wrap{
+  padding: 20px;
+}
+
+.flex-container{
+  display: flex;
+  background-color: transparent;
+  justify-content: space-evenly;
+  align-items: stretch;
+  flex-direction : row;
```
Comment (EmOlsz) [inline code/style.css:107]: There shouldn't be any space before colon

File: code/style.css [hunk 9]
```diff
@@ -1,7 +1,482 @@
...
+  flex-wrap: wrap;
+  padding-top: 3%;
```
Comment (EmOlsz) [inline code/style.css:109]: I see you really like to experiment with different units in paddings :D As before, `px` are the most proper ones for that

File: code/style.css [hunk 10]
```diff
@@ -1,7 +1,482 @@
...
+  color:white;
+  flex-basis : 100%; 
+}
+
+form{
+  display: flex;
+    flex-direction: column;
+}
+
+.text-Input{
```
Comment (EmOlsz) [inline code/style.css:132]: we shouldn't capitalize class names

File: code/style.css [hunk 11]
```diff
@@ -1,7 +1,482 @@
...
+  width: 99%;
+  border-radius: 13px;
+  border: none;
+  min-height: 2.5em;
+  font-size: 1.2em;
+}
+
+.text-Input:focus{
+  outline: none;
+  border-bottom: 4px solid#e89a74;
```
Comment (EmOlsz) [inline code/style.css:144]: you forgot space

File: code/style.css [hunk 12]
```diff
@@ -1,7 +1,482 @@
...
+  height: 19px;
+  width: 19px;
+  background-color: #f9f1f1;
+}
+
+.radiobutton-container input:checked ~.custom-radio{
+  background-color: #fff;
+}
+
+.custom-radio:after{
```
Comment (EmOlsz) [inline code/style.css:239]: Great job!

File: code/style.css [hunk 13]
```diff
@@ -1,7 +1,482 @@
...
+  top: 6px;
+  left: 6px;
+  width:7px;
+  height: 7px;
+  background-color: #2e5962;
+  border-radius: 50%;
+  display: none;
+}
+
+.radiobutton-container input:checked ~.custom-radio:after{
```
Comment (EmOlsz) [inline code/style.css:251]: And over here :)

File: code/style.css [hunk 14]
```diff
@@ -1,7 +1,482 @@
...
+  display: block;
+}
+
+.radiobutton-container input:focus + .custom-radio{
+  outline: #333 auto 4px;
```
Comment (EmOlsz) [inline code/style.css:256]: You don't actually need `4px` over here, it's working fine without it

File: code/style.css [hunk 15]
```diff
@@ -1,7 +1,482 @@
...
+.custom-checkbox:after{
+  content: "";
+  position: absolute;
+  width: 3px;
+  height: 6px;
+  border: solid #2e5962;
+  border-width: 0 3px 3px 0;
+  top: 5px;
+  left: 7px;
+  transform: rotate(45deg);
```
Comment (EmOlsz) [inline code/style.css:301]: 💪

File: code/style.css [hunk 16]
```diff
@@ -1,7 +1,482 @@
...
+  display: none;
+  background-color: #f9f1f1;
+  color: #333;
+  overflow: hidden;
+  border-bottom-left-radius: 4px;
+  border-bottom-right-radius: 4px;
+  font-size: 1em;
+}
+
+footer{
```
Comment (EmOlsz) [inline code/style.css:409]: Why not using class `.footer`?

**Review by EmOlsz — APPROVED** [review]
Great job Karolin, you made very nice project this week! First of all, whole site is working, there is a nice hero section, proper form and accordion. I like you design, especially border-bottom of inputs indicating when field is valid and when not, congrats! You've experimented a lot, I notice that.

Still, I have to mention about how your code looks like. Unfortunately, it's super unaesthetic. Please, consider checking it before pushing. Recruiters and companies are paying great attention to this kind of stuff and in your case everything else is completely fine, but because of that they could decide to hire somebody else.
Moreover, I'm not sure whether you understand every line of code you write. It's good that you use it, but try to deeply understand it every time.

Apart from that - congratulations, looking forward to your next project!

**Comment by karolinalbinsson** [comment]
> Great job Karolin, you made very nice project this week! First of all, whole site is working, there is a nice hero section, proper form and accordion. I like you design, especially border-bottom of inputs indicating when field is valid and when not, congrats! You've experimented a lot, I notice that.
> 
> Still, I have to mention about how your code looks like. Unfortunately, it's super unaesthetic. Please, consider checking it before pushing. Recruiters and companies are paying great attention to this kind of stuff and in your case everything else is completely fine, but because of that they could decide to hire somebody else.
> Moreover, I'm not sure whether you understand every line of code you write. It's good that you use it, but try to deeply understand it every time.
> 
> Apart from that - congratulations, looking forward to your next project!

Hi Max! Thank you for your extensive feedback. I'm happy with the result of this weeks work, and I tried really hard to think about how my code looks like. Yet it seems I am still being too sloppy. I have some follow up questions that I hope you can reply so I can improve. 

When you say that the code is "super unaesthetic" - is it because of indentation, spaces, line breaks and code that is commented out? Is it only that or is it also that I am using different units for e.g. paddings as you mention in the feedback. I'd like to understand a bit better what you mean with the aesthetics. Also, how can I show that I do understand the code I am writing? Should I include more comments?
//Karolin

**Comment by EmOlsz** [comment]
> Hi Max! Thank you for your extensive feedback. I'm happy with the result of this weeks work, and I tried really hard to think about how my code looks like. Yet it seems I am still being too sloppy. I have some follow up questions that I hope you can reply so I can improve.
> 
> When you say that the code is "super unaesthetic" - is it because of indentation, spaces, line breaks and code that is commented out? Is it only that or is it also that I am using different units for e.g. paddings as you mention in the feedback. I'd like to understand a bit better what you mean with the aesthetics. Also, how can I show that I do understand the code I am writing? Should I include more comments?
> //Karolin

Hi Karolin! Thank you for message :) Let me divide your questions :

1. **_When you say that the code is "super unaesthetic" - is it because of indentation, spaces, line breaks and code that is commented out? Is it only that or is it also that I am using different units for e.g. paddings as you mention in the feedback_** - It's mostly about indentation, spaces and line breaks stuff. It's just not consistent at all. Your code does work (and, actually, works super well, I'm very impressed by it), but looks bad. I'm completely sure it will take you 5 minutes per each project as the last step of development to clean it, try to do a habit out of it :)
1/2. Regarding different units in your code - since it's still Week 3 out of 20 weeks, I'm not yet super demanding when it comes to consistency of units, because I assume you want to play around with them a bit and try to understand all of them. With every project you will start to see that `px` is just the most convenient unit out there and at this point I just want to let it be said.
2. **_Also, how can I show that I do understand the code I am writing? Should I include more comments_** - This comment of mine was mostly addressed to your JavaScript file. Your code is actually quite unnecessary because it doesn't get a chance to be called - you've already added another validation directly in HTML (`required` attributes). In your JS file conditionals you've added `alert()` methods, but it's not possible to call them. That gives me clear information that you were not aware of that fact.

Just to be completely clear - consider these two (well, two and a half) comments as a warm encouragement to improve. I'm writing this especially because I see a lot of potential for being a good programmer in you

**Comment by karolinalbinsson** [comment]
Okay, thanks for the clarification! And thanks for the encouraging words :)

**Pull request closed without merging**